File: string_problems/DetermineColorOfChessboardSquare.js

// Problem link - https://leetcode.com/problems/determine-color-of-a-chessboard-square/description/?envType=problem-list-v2&envId=string

let squareIsWhite = function(coordinates) {
    // step 1 - extract character and number from coordinates
    let char = coordinates[0];
    let num = coordinates[1];

    // step 2 - find sum of both co-ordinates by converting the character into corresponding numeric value 
    let sum = (char.charCodeAt(0) - 'a'.charCodeAt(0) + 1) + Number(num);
    
    // step 3 - check whether sum is odd or even and return true and false accordingly
    if(sum % 2 === 0){
        // return false for even sum
        return false;
    }else{
        // return true for odd sum
        return true;
    }
};

// const coordinates = "a1";

// const coordinates = "b3";

const coordinates = "c7";
console.log(squareIsWhite(coordinates));

File: string_problems/DetermineColorOfChessboardSquare.java

// Problem link - https://leetcode.com/problems/determine-color-of-a-chessboard-square/description/?envType=problem-list-v2&envId=string

package string_problems;

public class DetermineColorOfChessboardSquare {
    public static boolean squareIsWhite(String coordinates) {
        // step 1 - extract character and number from coordinates
        int charVal = coordinates.charAt(0) - 'a' + 1;
        int num = coordinates.charAt(1) - '0';

        // step 2 - find sum of both co-ordinates
        int sum = charVal + num;

        // step 3 - check whether sum is odd or even and return true and false accordingly
        if(sum % 2 == 0){
            // return false for even sum
            return false;
        }else{
            // return true for odd sum
            return true;
        }
    }

    public static void main(String[] args) {
        // String coordinates = "a1";

        // String coordinates = "h3";

        String coordinates = "c7";
        System.out.println(squareIsWhite(coordinates));
    }
}